message-queue-lib/src/cmr/message_queue/topic/local_topic.clj

```diff
@@ -45,11 +63,44 @@
 
   topic-protocol/Topic
   (subscribe
-    [_this subscription]
-    (swap! subscription-atom conj subscription))
+   [this subscription]
+   ;; to speed up development startup, the setup call is here and setup checks first to see if it is already setup.
+   ;; Otherwise on startup the system would have to wait for the elasticmq to start before it could continue with setting
+   ;; up the database slowing down all the tests.
+   (setup-infrastructure this)
+   (let [metadata (:metadata subscription)
+         sqs-client (queue/create-sqs-client (config/sqs-server-url))
```

Comment:
Does an SQS client need to be made for each subscription? Is it like a different interface to the SQS service for each subscription?

Reply:
I never figured out where the underlying http connection (assuming its http) lived in the AWS libraries.  I thought that it would live in the client, so I wanted to limit those and keep them open for quicker communication, but if that is not the case, then we don't need it in the subscription. It is possible that a queue could live in a different place from other queues, but when the subscription is set up all of the clients are set to use the local queue manager.